autogen/agentchat/contrib/agent_optimizer.py

```diff
@@ -199,14 +199,17 @@ def __init__(
         self._failure_functions_performance = []
         self._best_performance = -1
 
-        config_list = autogen.config_list_from_json(
-            config_file_or_env,
-            file_location=config_file_location,
-            filter_dict={"model": [self.optimizer_model]},
+        assert isinstance(llm_config, dict), "llm_config must be a dict"
+        llm_config = copy.deepcopy(llm_config)
+        self.llm_config = llm_config
+        if self.llm_config in [{}, {"config_list": []}, {"config_list": [{"model": ""}]}]:
+            raise ValueError(
+                "When using OpenAI or Azure OpenAI endpoints, specify a non-empty 'model' either in 'llm_config' or in each config of 'config_list'."
+            )
```

Comment:
would this work as expected? what if `self.llm_config = {"config_list": [], "seed": 42}`?

Reply:
Maybe expose, improve and reuse `_validate_llm_config` in ConversableAgent

Reply:
> Maybe expose, improve and reuse `_validate_llm_config` in ConversableAgent

llm_config requirement here is different from it in ConversableAgent. So we can not expose and re-use it here

Reply:
> would this work as expected? what if `self.llm_config = {"config_list": [], "seed": 42}`?

That is a common problem in both ConversableAgent and here. I will make an issue and another PR to address it. We can merge this PR for now.